cmake/Modules/UseOptimization.cmake

```diff
@@ -64,8 +64,8 @@ if (CXX_COMPAT_GCC)
   # use these options for release builds - full optimization
   add_options (ALL_LANGUAGES "${_prof_RELEASE}" ${_opt_rel} ${_opt_flags})
   option(WITH_NDEBUG "Disable asserts in release mode" ON)
-  if(WITH_NDEBUG)
-    add_options (ALL_LANGUAGES "${_prof_RELEASE}" -DNDEBUG)
+  if(NOT WITH_NDEBUG)
+    add_options (ALL_LANGUAGES "${_prof_RELEASE}" -UNDEBUG)
```

Comment:
Shouldn't we rather just remove -DNDEBUG from flags using `list(REMOVE_ITEM flags -DNDEBUG})` in a macro remove_options or similar?

Reply:
maybe. i did not want to deal with the snake pit that is flags variables (they are usually, but not always, a string rather than a proper list).